src/matcha_ml/cli/provision.py

```diff
@@ -84,40 +84,41 @@ def provision_resources(
         remote_state_manager.provision_state_storage(location, prefix)
 
     with remote_state_manager.use_lock():
-        # create a runner for provisioning resource with Terraform service.
-        template_runner = AzureTemplateRunner()
+        with remote_state_manager.use_remote_state():
+            # create a runner for provisioning resource with Terraform service.
+            template_runner = AzureTemplateRunner()
```

Comment:
Do we require all these upto line 110 in the context manager itself?

Reply:
As above, I assumed that we'd need the template runners within the context so that they are caught by the upload call on the first provision.